storage/src/main/java/zipkin2/storage/kafka/KafkaStoreServerSupplier.java

```diff
@@ -0,0 +1,328 @@
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.annotation.ConsumesJson;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.Param;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+import org.apache.kafka.streams.errors.InvalidStateStoreException;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.QueryableStoreTypes;
+import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+import org.apache.kafka.streams.state.ReadOnlyWindowStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin2.DependencyLink;
+import zipkin2.Span;
+import zipkin2.codec.DependencyLinkBytesEncoder;
+import zipkin2.codec.SpanBytesEncoder;
+import zipkin2.internal.DependencyLinker;
+import zipkin2.storage.QueryRequest;
+import zipkin2.storage.kafka.internal.Traces;
+
+import static zipkin2.storage.kafka.streams.DependencyStoreTopologySupplier.DEPENDENCIES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.AUTOCOMPLETE_TAGS_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.REMOTE_SERVICE_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SERVICE_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SPAN_IDS_BY_TS_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SPAN_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.TRACES_STORE_NAME;
+
+/**
+ * Server to enable access to local stores.
+ * <p>
+ * Given the partitioned nature of local stores, a RPC layer is required to allow accessing
+ * distributed state. This component exposes access to local state via Http call from {@link
+ * KafkaSpanStore}
+ *
+ * @since 0.6.0
+ */
+public class KafkaStoreServerSupplier implements Supplier<Server> {
+  static final Logger LOG = LoggerFactory.getLogger(KafkaStoreServerSupplier.class);
+  static final Gson GSON = new Gson();
+
+  final KafkaStorage storage;
+  final long minTracesStored;
+  final int httpPort;
+  final PrometheusMeterRegistry prometheusRegistry;
+
+  KafkaStoreServerSupplier(KafkaStorage storage) {
+    this.storage = storage;
+
+    this.minTracesStored = storage.minTracesStored;
+    this.httpPort = storage.httpPort;
+    this.prometheusRegistry = storage.prometheusRegistry;
+  }
+
+  @Override public Server get() {
```

Comment:
basically change this to `Consumer<ServerBuilder>` and don't set things like port. namespace the api under a variable or /storage-kafka/

Reply:
I'm probably missing something huge, but why do we add a separate API endpoint for scatter-gather? Shouldn't we use the same zipkin API but have the `Storage` implemented using scatter-gather, which I think the above files are doing?